hcipy/aperture/realistic.py

```diff
@@ -1079,3 +1079,249 @@ def func(grid):
+			[-1.89978486, 1.97305809],
+			[-1.52770672, 1.32197434],
+			[-1.90871672, 0.66204566],
+			[-2.65861086, 0.65873291],
+			[-3.030786, 1.30987266]]),
+		'C1-8': np.array([
+			[0.765201, 2.627516],
+			[1.517956, 2.629178],
+			[1.892896, 1.976441],
+			[1.521076, 1.325812],
+			[0.765454, 1.325807],
+			[0.387649, 1.980196]]),
+		'C2-10': np.array([
+			[2.6580961, 0.651074495],
+			[3.03591294, 5.42172989e-07],
+			[2.65809612, -0.651075523],
+			[1.90872487, -0.654384457],
+			[1.53090954, 8.90571587e-07],
+			[1.90872454, .654384118]]),
+		'C3-12': np.array([
+			[1.8928951, -1.97644151],
+			[1.51795694, -2.62917746],
+			[0.76520012, -2.62751652],
+			[0.38764887, -1.98019646],
+			[0.76545554, -1.32580611],
+			[1.52107554, -1.32581188]]),
+		'C4-14': np.array([
+			[-0.765201, -2.627516],
+			[-1.517956, -2.629178],
+			[-1.892896, -1.976441],
+			[-1.521076, -1.325812],
+			[-0.765454, -1.325807],
+			[-0.387649, -1.980196]]),
+		'C5-16': np.array([
+			[-2.6580961, -.651074495],
+			[-3.03591294, -5.42172990e-07],
+			[-2.65809612, .651075523],
+			[-1.90872487, .654384457],
+			[-1.53090954, -8.90571587e-07],
+			[-1.90872454, -.654384118]]),
+		'C6-18': np.array([
+			[-1.8928951, 1.97644151],
+			[-1.51795694, 2.62917746],
+			[-0.76520012, 2.62751652],
+			[-0.38764887, 1.98019646],
+			[-0.76545554, 1.32580611],
+			[-1.52107554, 1.32581188]]),
+	}
+
+	strut_parameters = {
+		"strut1": np.array([
+			[-0.05301375, -0.0306075],
+			[1.59698625, -2.88849133],
+			[1.70301375, -2.82727633],
+			[0.05301375, 0.0306075],
+			[-0.05301375, -0.0306075]]),
+		"strut2": np.array([
+			[-0.05301375, 0.0306075],
+			[-1.70301375, -2.82727633],
+			[-1.59698625, -2.88849133],
+			[0.05301375, -0.0306075],
+			[-0.05301375, 0.0306075]]),
+		"strut3": np.array([
+			[5.94350000e-02, -1.45573765e-17],
+			[5.94350000e-02, 3.30000000e+00],
+			[-5.94350000e-02, 3.30000000e+00],
+			[-5.94350000e-02, 1.45573765e-17],
+			[5.94350000e-02, -1.45573765e-17]]),
+		"strut3_bumps": np.array([
+			[0.059435, 0.666],
+			[0.059435, 2.14627],
+			[0.082595, 2.14627],
+			[0.082595, 2.3645],
+			[0.059435, 2.3645],
+			[0.059435, 2.48335],
+			[0.069795, 2.48335],
+			[0.069795, 2.54445],
+			[0.059435, 2.54445],
+			[0.059435, 3.279674],
+			[-0.059435, 3.279674],
+			[-0.059435, 2.54445],
+			[-0.069795, 2.54445],
+			[-0.069795, 2.48335],
+			[-0.059435, 2.48335],
+			[-0.059435, 2.3645],
+			[-0.082595, 2.3645],
+			[-0.082595, 2.14627],
+			[-0.059435, 2.14627],
+			[-0.059435, 0.666]])
+	}
+
+	if normalized:
+		for val in segment_parameters.values():
+			val /= pupil_diameter
+
+		for val in strut_parameters.values():
+			val /= pupil_diameter
+
+		pupil_diameter = 1
+
+	segments = [irregular_polygon_aperture(vertices) for vertices in segment_parameters.values()]
+	struts = [make_obstruction(irregular_polygon_aperture(vertices)) for vertices in strut_parameters.values()]
+
+	def func(grid):
+		ota = sum((segment(grid) for segment in segments))
+
```

Comment:
what is ota? It is probably better to use a descriptive variable name.

Reply:
optical telescope assembly. I agree with not having acronyms, so I'll change it.

Reply:
ok.